### PR TITLE
fix: Add pathType to create networking/v1 ingress

### LINKED ISSUE #2144: Unable to create ingress route inside Google Kubernetes Engine
**Fission/Kubernetes version**

<pre>
$ fission --version

client:
  fission/core:
    BuildDate: "2021-08-02T11:44:22Z"
    GitCommit: f4702fe066c48e05cd5c093311fe95e4a6a971a4
    Version: 1.14.0
server:
  fission/core:
    BuildDate: "2021-08-02T09:30:56Z"
    GitCommit: f4702fe066c48e05cd5c093311fe95e4a6a971a4
    Version: 1.14.0

$ kubectl version

Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.3", GitCommit:"ca643a4d1f7bfe34773c74f79527be4afd95bf39", GitTreeState:"clean", BuildDate:"2021-07-15T21:04:39Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.8-gke.900", GitCommit:"28ab8501be88ea42e897ca8514d7cd0b436253d9", GitTreeState:"clean", BuildDate:"2021-06-30T09:23:36Z", GoVersion:"go1.15.13b5", Compiler:"gc", Platform:"linux/amd64"}
</pre>

**Kubernetes platform (e.g. Google Kubernetes Engine)**
Google Kubernetes Engine

**Describe the bug**
```bash
fission route create --url /hello --function hello --createingress
```

Trying to create route with `--createingress` passes with success from Fission CLI but when i open `router` pod inside Google Cloud Engine i can see in the logs that error message appears  `error: Ingress.extensions "daa2b59c-d841-4a47-824e-7a460e810aee" is invalid: spec.rules[0].http.paths[0].pathType: Required value: pathType must be specified`

Full error message:

```
2021-08-03 13:13:16.955 EEST{caller: router/ingress.go:48, error: Ingress.extensions "daa2b59c-d841-4a47-824e-7a460e810aee" is invalid: spec.rules[0].http.paths[0].pathType: Required value: pathType must be specified, level: error, logger: triggerset.http_trigger_set, msg: failed to create ingress, ts: 2021-08-03T10:13:16.955Z…
```
Screenshot from 2021-08-03 13-23-37

**To Reproduce**

Create Google Kubernetes Cluster and try to create route with ingress in mind then go to router pod and watch for logs

**Expected result**
To not have error inside router and to create ingress route

**Additional Notes**

I have installed `ingress` using the link provided in the [Documentation](https://kubernetes.github.io/ingress-nginx/deploy/#gce-gke) especially for GKE 

Here is the output of `kubectl get all -n ingress-nginx`

```
NAME                                           READY   STATUS      RESTARTS   AGE
pod/ingress-nginx-admission-create-wc6v2       0/1     Completed   0          88m
pod/ingress-nginx-admission-patch-9gwhl        0/1     Completed   1          88m
pod/ingress-nginx-controller-b65df6fbb-r2x7f   1/1     Running     0          88m

NAME                                         TYPE           CLUSTER-IP    EXTERNAL-IP   PORT(S)                      AGE
service/ingress-nginx-controller             LoadBalancer   10.104.6.69   **.**.**.**   80:30248/TCP,443:31316/TCP   88m
service/ingress-nginx-controller-admission   ClusterIP      10.104.6.1    <none>        443/TCP                      88m

NAME                                       READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/ingress-nginx-controller   1/1     1            1           88m

NAME                                                 DESIRED   CURRENT   READY   AGE
replicaset.apps/ingress-nginx-controller-b65df6fbb   1         1         1       88m

NAME                                       COMPLETIONS   DURATION   AGE
job.batch/ingress-nginx-admission-create   1/1           6s         88m
job.batch/ingress-nginx-admission-patch    1/1           7s         88m
```

### PR DESCRIPTION
**Pull request by chenjr0719**
In #2144 , the error is because of lacking `pathType` when creating ingress. This PR simply add `pathType` to the ingress spec.

fix #2144

### CODE REVIEW

**Comment by sanketsudake** [comment]
@chenjr0719 Thanks for the fix, it looks good to me. Would you be able to check unit tests also as they are failing ?

**Comment by sanketsudake** [comment]
I made a minor change in code and fixes in unit tests. @chenjr0719 Let me know if you can consider this patch, or if I can push this commit to current PR.
```
commit a93107b08574fc71ae3175346afefac3ece38773
Author: Sanket Sudake <sanketsudake@gmail.com>
Date:   Thu Aug 19 08:57:37 2021 +0530

    Add fixes for the unit tests

    Signed-off-by: Sanket Sudake <sanketsudake@gmail.com>

diff --git a/pkg/router/util/util.go b/pkg/router/util/util.go
index 3d886f3f..742c4ea4 100644
--- a/pkg/router/util/util.go
+++ b/pkg/router/util/util.go
@@ -42,7 +42,6 @@ func GetIngressSpec(namespace string, trigger *fv1.HTTPTrigger) *v1.Ingress {
 	}

 	var ingTLS []v1.IngressTLS
-	var pathType v1.PathType = "ImplementationSpecific"
 	if len(trigger.Spec.IngressConfig.TLS) > 0 {
 		ingTLS = []v1.IngressTLS{
 			{
@@ -54,6 +53,7 @@ func GetIngressSpec(namespace string, trigger *fv1.HTTPTrigger) *v1.Ingress {
 		}
 	}

+	var pathType v1.PathType = v1.PathTypeImplementationSpecific
 	ing := &v1.Ingress{
 		ObjectMeta: metav1.ObjectMeta{
 			Labels: GetDeployLabels(trigger),
diff --git a/pkg/router/util/util_test.go b/pkg/router/util/util_test.go
index ac3762d8..185a6001 100644
--- a/pkg/router/util/util_test.go
+++ b/pkg/router/util/util_test.go
@@ -31,6 +31,7 @@ type args struct {
 		ingressNS string
 		trigger   *fv1.HTTPTrigger
 	}
+	var pathType v1.PathType = v1.PathTypeImplementationSpecific
 	tests := []struct {
 		name string
 		args args
@@ -84,7 +85,8 @@ type args struct {
 													},
 												},
 											},
-											Path: "/foo/bar",
+											Path:     "/foo/bar",
+											PathType: &pathType,
 										},
 									},
 								},
@@ -145,7 +147,8 @@ type args struct {
 													},
 												},
 											},
-											Path: "/foo/bar",
+											Path:     "/foo/bar",
+											PathType: &pathType,
 										},
 									},
 								},
@@ -204,7 +207,8 @@ type args struct {
 													},
 												},
 											},
-											Path: "/foo/bar",
+											Path:     "/foo/bar",
+											PathType: &pathType,
 										},
 									},
 								},
@@ -263,7 +267,8 @@ type args struct {
 													},
 												},
 											},
-											Path: "/foo/{bar}",
+											Path:     "/foo/{bar}",
+											PathType: &pathType,
 										},
 									},
 								},
@@ -322,7 +327,8 @@ type args struct {
 													},
 												},
 											},
-											Path: "/foo/{bar}",
+											Path:     "/foo/{bar}",
+											PathType: &pathType,
 										},
 									},
 								},
@@ -382,7 +388,8 @@ type args struct {
 													},
 												},
 											},
-											Path: "/foo/bar",
+											Path:     "/foo/bar",
+											PathType: &pathType,
 										},
 									},
 								},
@@ -441,7 +448,8 @@ type args struct {
 													},
 												},
 											},
-											Path: "/foo/bar",
+											Path:     "/foo/bar",
+											PathType: &pathType,
 										},
 									},
 								},
@@ -512,7 +520,8 @@ type args struct {
 													},
 												},
 											},
-											Path: "/foo/bar",
+											Path:     "/foo/bar",
+											PathType: &pathType,
 										},
 									},
 								},
```

File: pkg/router/util/util.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func GetIngressSpec(namespace string, trigger *fv1.HTTPTrigger) *v1.Ingress {
 	}
 
 	var ingTLS []v1.IngressTLS
+	var pathType v1.PathType = "ImplementationSpecific"
```
Comment (sanketsudake) [inline pkg/router/util/util.go:45]: We can avoid hardcoding with direct reference to the constant.
```
var pathType v1.PathType = v1.PathTypeImplementationSpecific
```

Please refer to comment https://github.com/fission/fission/pull/2164#issuecomment-901582182

**Comment by chenjr0719** [comment]
Thank you @sanketsudake . Please push your commit directly, I think they are better than mine. 😅

**Pull request merged**